### DISCUSSION TITLE
YAML tags entered incorrectly when using `git` sync (Stable and Beta)

### DISCUSSION THREAD

**Discussion by RyanGreenup** [requarks/wiki#3525] [Error / Bug Report]
When using the `git` integration, in `2.3.81`, tags are added to the *YAML* header like this:

```
---
tags: tag1, tag2, tag3
---
# My MD Note
content blah...
```

From what I understand of the [YAML standard](https://yaml.org/spec/1.2/spec.html) tags should be added like this:

```
---
tags: [tag1, tag2, tag3]
---
# My MD Note
content blah...
```

~~Also in the new beta `2.4.75` tags are not entered into the YAML at all.~~

I propose that:

1.  `tags` should get entered into the YAML on `git` sync,     AND
2.  tags should work in both directions,     AND
3.  The `yaml` property should be modified to be be a list like 
  * `tags: [tagA, tagB, tagC]`

This would bring the following benefits:

1. Compatability with existing libraries

    *  for example in **_R_** there is the `rmarkdown::yaml_front_matter()` function, this allows me to extract tags from markdown files, but without the square brackets it returns an error.
    *  The exact same same goes for ` yaml-front-matter ` a package in *npm*
2. Work with [*Notable*](https://github.com/notable/notable) which also expects the square brackets.

Better Cross compatability generally would be desirable.

---
**related**: if the YAML is changed into an inconsistent format through say *GitHub* I had some issues with getting the sync to work properly until I did `git revert`, *WikiJS* should give a notification that the YAML has been entered incorrectly and do something like make a seperate branch or force the user to merge changes or something like that, the silent failure was really really confusing to work through and I was anticipating the error occuring because I was testing it.

**Comment by RyanGreenup** [comment]
just tested with `STABLE - 2.4.105` this issue persists, tags are entered into the *YAML* without square brackets same as  `2.3.81`. The ideal solution would be to modify them to insert `[as, a, list] ` with bidirectional sync of the tags so as to be consistent with other Markdown *YAML* headers.

**Comment by tionis** [comment]
I second this, it's a really annoying breach of the standard.
